tokio-test/src/io.rs

```diff
@@ -453,6 +453,16 @@ impl AsyncWrite for Mock {
     }
 }
 
+/// Ensures that Mock isn't dropped with data "inside".
+impl Drop for Mock {
+    fn drop(&mut self) {
+        self.inner.actions.iter().for_each(|a| match a {
```

Comment:
we should probably add
```suggestion
        if std::thread::panicking() {
            return;
        }
        self.inner.actions.iter().for_each(|a| match a {
```
to avoid double panics. Panicking during a panic results in the process getting killed immediately with a SIGILL, rather than printing a panic message, so it would make the test much harder to debug if a different assertion failed.